### PR TITLE
Updated not working swiftyjson and add collection view to display albums when clicking on artist

### PR DESCRIPTION
**Pull request by DanaDrimba**

### CODE REVIEW

File: LastFM/ArtistAlbumsCollectionViewController.swift [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+
+    override func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
+        return self.items.count
+    }
+
+    override func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
+        
+        let cell = collectionView.dequeueReusableCell(withReuseIdentifier: reuseIdentifier, for: indexPath) as! AlbumCoverCollectionViewCell
+    
+        let album = self.items[indexPath.row]
```
Comment (flaviu88) [inline LastFM/ArtistAlbumsCollectionViewController.swift]: You aren't required to explicitly  use self before accessing a variable from the same class. You can just use: items[indexPath.row] . The only place where you need to explicitly add self before a variable is in a closure.

File: LastFM/ArtistAlbumsCollectionViewController.swift [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+        cell.backgroundColor = UIColor.white
+        if let url = NSURL(string: album.pictureURL) {
```
Comment (flaviu88) [inline LastFM/ArtistAlbumsCollectionViewController.swift]: You can combine this two if's on a same line like this: if let url = NSURL(string: album.pictureURL), let data = NSData(contentsOf: url as URL) {

File: LastFM/FirstViewController.swift [hunk 3]
```diff
@@ -43,7 +37,7 @@ class FirstViewController: UIViewController, UITableViewDataSource, UITableViewD
                     self.items.append(LastFMArtist(json: entry))
                 }
                 DispatchQueue.main.async(execute: {
-                    self.tableView.reloadData()
+                    self.topArtistsTableView.reloadData()
```
Comment (flaviu88) [inline LastFM/FirstViewController.swift]: See the comment above about using self inside a closure.

File: LastFM/SecondViewController.swift [hunk 4]
```diff
@@ -47,24 +47,22 @@ class SecondViewController: UIViewController, UITableViewDataSource, UITableView
     }
     
     func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
-        var cell = tableView.dequeueReusableCell(withIdentifier: "CELL")
-        
-        if cell == nil {
-            cell = UITableViewCell(style: UITableViewCellStyle.subtitle, reuseIdentifier: "CELL")
-        }
+
+        let cell = tableView.dequeueReusableCell(withIdentifier: "CELL", for: indexPath)
```
Comment (flaviu88) [inline LastFM/SecondViewController.swift]: Add a constant for the cell identifier.

File: LastFM/ArtistAlbumsCollectionViewController.swift [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        //get the parameter from the previous view controller
+        self.getTopAlbums(forArtist: "Coldplay")
+    }
+    
+    func getTopAlbums(forArtist: String) {
+        
+        DataManager.sharedInstance.getTopAlbums(forArtist: forArtist) { (json:JSON) in
```
Comment (DanaDrimba) [inline LastFM/ArtistAlbumsCollectionViewController.swift]: xCode says it need them parenthesis

**Pull request merged**